src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -2661,6 +2661,7 @@ ExtractInsertPartitionKeyValue(Query *query)
 	Var *partitionColumn = NULL;
 	TargetEntry *targetEntry = NULL;
 	Const *singlePartitionValueConst = NULL;
+	Expr *targetExpression = NULL;
```

Comment:
minor: if you make this a `Node *` you can avoid a cast

Reply:
done